src/cfnlint/runner.py

```diff
@@ -434,8 +434,14 @@ def cli(self) -> None:
             sys.exit(0)
 
         if not self.config.templates:
+            self.config.parser.print_help()
+
+            # If templates were specified, but didn't create any file matches, print out message and exit
+            template = self.config.get_input_templates
+            if template:
+                print(f"\nNo matching files found for template arguments: {template}")
```

Comment:
I actually wonder if this should just be a raised exception like InvalidRegionException is configured (InvalidTemplateException ?). or a LOGGER.debug statement?

A random print here may cause issues with people parsing the output as json or some of other formats.

Reply:
in hindsight, I can the print to the `_glob_filenames` function inside `config.py` and change it to a `warn` message?

Currently if no templates are found, it'll output the usage - What would you like the behaviour of `cfn-lint` to be in this case?

```diff
    def _glob_filenames(self, filenames: Sequence[str]) -> list[str]:
        # handle different shells and Config files
        # some shells don't expand * and configparser won't expand wildcards
        all_filenames = []

        for filename in filenames:
            add_filenames = glob.glob(filename, recursive=True)

            if isinstance(add_filenames, list):
                all_filenames.extend(add_filenames)
            else:
                all_filenames.append(filename)
+      if filenames and not all_filenames:
+          LOGGER.warn(f"No matching files found for {filenames}")
        return sorted(list(map(str, map(Path, all_filenames))))
```

Reply:
Yea, good question.  Thinking through this.  Always like some outside feedback here to @benbridts ?

I'm not sure we should print the usage as they technically used it correctly they just provided a valid path doesn't have any files (or a bad path).  

This will also depend on a little by the shell a bad path will get caught by ZSH but not in bash (or at least my configuration of both shells).

This is what I'm seeing now.  I don't think the E0000 makes sense here either.  I'm thinking we should send the logging message and then continue gracefully.  If the list is empty the list is empty and we just exit.  
```
2024-08-19 10:28:55,415 - cfnlint.decode.decode - ERROR - Template file not found: ~/Downloads/templates1/**/*.yaml
E0000 Template file not found: ~/Downloads/templates1/**/*.yaml
~/Downloads/templates1/**/*.yaml:1:1
```

Reply:
In the latest changes, I've added a debug statement:

```
$ cfn-lint -D "cfn/**/*.y*ml"
2024-08-21 10:52:56,047 - cfnlint - DEBUG - List of Cloudformation Templates to lint: [] from ['cfn/**/*.y*ml']
```